tests/iio_writedev.c

```diff
@@ -421,6 +421,7 @@ int main(int argc, char **argv)
 	}
 
 	sample_size = iio_device_get_sample_size(dev);
+	bool num_samples_set = num_samples > 0;
```

Comment:
can `num_samples` be negative?
if so, maybe we should add a check in another patch and exit early;
maybe here:
```
		case 's':
			num_samples = atoi(optarg);
			break;
```
the reason i am asking, is because this is checking `num_samples > 0;`

in any case, adding an extra variable `num_samples_set` seems to be extra;
you can probably remove it & and just check  `if (num_samples)`  or `if (!num_samples)` where needed

Reply:
num_samples can't be negative; it is of type size_t, which is an unsigned data type. 
I have added the num_samples_set variable because when num_samples becomes 0, these lines `if (num_samples && len > num_samples * sample_size)
				len = num_samples * sample_size;`
will be ignored, letting len = buffer_size. Therefore, from now on, the program will behave like no num_samples was provided and will push all the available samples from stdin (after pushing the requested number of samples).

Reply:
so, how about `bool num_samples_zero = (num_samples != 0)` ? might be more intuitive.

Reply:
Wouldn't it be more intuitive this way:
`bool num_samples_zero = (num_samples == 0)`, num_samples_zero being true if num_samples is 0? 
On the other hand, wouldn't it be a little confusing this way, in the situation in which num_samples becomes 0 (after pushing some samples)? I think on the situation in which someone is debugging the program and expecting num_samples_zero to become true when num_samples becomes 0.